### PR TITLE
fix: Remove `getUser` in callback handler

### LINKED ISSUE #447: Accessing state after signin
I am sending some state with my signin request as wish to access it again once the signin is complete 

```typescrpt
export const SimpleLogin: React.FC = (): JSX.Element => {
  const auth = useAuth();
  
  return (
    
      <div>
        {auth?.userData && (
         <>{auth?.userData.profile.name || ''}</>
        )}
        {!auth?.userData && (
          <button onClick={() => auth?.signIn({ state: { targetUrl: window.location.pathname } })}>
            Login
          </button>
        )}
      </div>
  );
};
```

i need to access this state after login so that i can redirect to the correct path in my application, however state is undefined on the User object in the onSignIn event 

```typescrpt
<AuthProvider
      authority='someAuthority'
      clientId='someclientid'
      automaticSilentRenew
      responseType="code"
      autoSignIn={false}
      onSignIn={(user) => {
        console.log(user.state);
         // user.state is undefined
      }}
      scope="openid profile email "
      redirectUri={window.location.origin}
    >
      <RestOfMyApp />
    </AuthProvider>
```
How can i get access to the state after signin. I can see that the state is being included in the query paramaters along with the code on the redirect url, but cant work out how to access it, any help would be greatly welcomed

I noticed #36 allows the state to be sent to the authorizer server so presume it is getting through? Bu then not accessible when try to access onSignIn? @johnkors  @cobraz  any thoughts?

thanks, Ed

### PR DESCRIPTION
**Pull request by ed-sparkes**
Update to the signincallback flow to make state returned from oidc-client available to the onSignIn event

fixes #447

### CODE REVIEW

**Comment by ed-sparkes** [comment]
not sure what to do about the semantic pull request i renamed the PR to be prefix with fix: but not sure how to "Squash and Merge"

File: src/AuthContext.tsx [hunk 1]
```diff
@@ -85,8 +85,7 @@ export const AuthProvider: FC<AuthProviderProps> = ({
        * Check if the user is returning back from OIDC.
        */
       if (hasCodeInUrl(location)) {
-        await userManager.signinCallback();
-        const user = await userManager.getUser();
+        const user = await userManager.signinCallback();
```
Comment (cobraz) [inline src/AuthContext.tsx:88]: Okey, so `signinCallback` returns the same as `getUser`?

**Comment by cobraz** [comment]
> not sure what to do about the semantic pull request i renamed the PR to be prefix with fix: but not sure how to "Squash and Merge"

Don't worry about that, I can fix that :)

**Review by cobraz — APPROVED** [review]
Thank you so much for contributing to this project!

**Comment by ed-sparkes** [comment]
sweet, its merged :) how does this get upversioned and released to npm ?

**Comment by cobraz** [comment]
> sweet, its merged :) how does this get upversioned and released to npm ?

Already done 👇

> 🎉 This PR is included in version 1.1.6 🎉
> 
> The release is available on:
> 
> * [npm package (@latest dist-tag)](https://www.npmjs.com/package/oidc-react/v/1.1.6)
> * [GitHub release](https://github.com/bjerkio/oidc-react/releases/tag/v1.1.6)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

**Pull request merged**